### ISSUE TITLE
Contextual toolbar config

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-ui#211]
I see that now contextual toolbar has a different config that other toolbars:

https://github.com/ckeditor/ckeditor5-ui/blob/66a30b1498cf52864d550a98d77eca3b2239ff4d/src/toolbar/contextual/contextualtoolbar.js#L82

I understand that one may want to have part of the buttons in the contextual toolbar and part in the sticky toolbar, but on the other had if the contextual toolbar is the only one, the config option should be the same. This is why I think that it should first look for the custom config and if it does not find, use the default one. In fact, we could make all toolbars working this way. Thanks to it you would be able to have:
```js
{
   toolbar: {
      contextual: [ 'bold', 'italic' ],
      sticky: [ 'image' ]
   }
}
```

Or if you have only a single toolbar or all toolbars should have the same buttons you could do:
```js
{
   toolbar: [ 'bold', 'italic', 'image' ]
}
```

**Comment by oleq** [comment]
I'm not so sure about it. In fact, I think we didn't consider that the toolbar might have to have a dynamic configuration (?), e.g. to support https://github.com/ckeditor/ckeditor5/issues/402. A single `CT` instance should display the general toolbar (bold, italic, etc.) but if the image is selected, it should immediately switch to the image context and display the right set of buttons. Gotta think about it more and figure out the right approach to support this.

**Comment by pjasiun** [comment]
I'm not talking about the image contextual options. I'm talking about having part of the buttons in one toolbar and part in the other. For instance, inline options in the contextual toolbar and block options in another toolbar, like slack editor do.

**Comment by fredck** [comment]
I think that the current proposal is that there is one "main toolbar", which is the one that is configured with `toolbar`. In the case of the contextual toolbar, this is the one used when a selection is made.

Then, you can have dozens of other features which also have toolbars, like "image" and also the "block options" that you're talking about. It would be then up to those features to handle their configuration and their toolbars.

The above is in fact what you see today with the Image feature. We can see it's toolbar configuration in the classic build:
https://github.com/ckeditor/ckeditor5-build-classic/blob/131cbe76b05b49c5b4df5d197459bdc08cf5a9b5/config-editor.js#L21-L23

It means that it would be enough to create a custom "block options" plugin which would do a job much similar to the Image feature.

All this should not be left to the contextual toolbar itself to handle, because otherwise there would be two different implementations - one for the contextual toolbar and another for other toolbar approaches, like the classic editor. So a single solution that satisfies both cases is the best approach.

**Comment by Reinmar** [comment]
I agree with @fredck and this is how we implemented the `editor-balloon-toolbar` – it just uses the `config.toolbar` option.

However, one thing is ugly now – how's `editor-balloon-toolbar` configuring the `ContextualToolbar` (which is a plugin)? Through `config.contextualToolbar`. Why? Because if we'd make `ContextualToolbar` read directly from `config.toolbar` the plugin would not be usable in editors which use `config.toolbar` for their main toolbar (like the classic editor).

This should be organised in some better way. I'm not sure now how but there should be a couple of options.
